src/info/git/mod.rs

```diff
@@ -172,20 +174,29 @@ type ChurnPair = (NumberOfCommitsByFilepath, usize);
 
 fn get_churn_channel(
     repo: &gix::Repository,
+    mailmap: &gix::mailmap::Snapshot,
+    bot_regex_pattern: &Option<MyRegex>,
```

Comment:
Passing `&Option<T>` isn't idiomatic, but `Option<&T>` is. That requires `.as_ref()` on the call-site. This is likely the work of an IDE and refactoring tools.

It's a minor detail though and one doesn't have to care, it also seems to be done in other places that aren't touched here.

Reply:
Is there perhaps a blanket `AsRef` implementation that can be used? Something like this?
```rust
bot_regex_pattern: impl AsRef<Option<MyRegex>>
```
(also perhaps `MyRegex` should implement `AsRef<Regex>`)

Reply:
I don't think that adding generics to method signatures would do more than allowing to avoid manual `.as_ref()`, which I'd be more than happy to do for clear compiler errors.
But in the end, it's probably Ok the way it is as it's most natural for IDEs. It's surprising clippy is happy with it though.

Reply:
TBH looking at this more it is kind of odd IMO to have functions that don't do anything significant unless that option is `Some`.
Basically instead of this
```rust
fn foo(ref: &Option<T1>) -> T2 {
    if bar(ref) {
        do_thing()
    } else {
        do_other_thing()
    }
}

fn bar(ref: &Option<T1>) -> bool {
    let thing = do_calculation();
    baz(thing, ref)
}

fn baz(thing: Thing, ref: &Option<T1>) -> bool {
    ref.as_ref().map_or(false, |t| t.process(thing))
}
```

I think it should be more like this:
```rust
fn foo(ref: &Option<T1>) -> T2 {
    if ref.as_ref().unwrap_or(false, bar) {
        do_thing()
    } else {
        do_other_thing()
    }
}

fn bar(ref: &T1) -> bool {
    let thing = do_calculation();
    ref.process(thing)
}
```

So, like you said, move the `.as_ref` call higher up in the "layers" of functions, but while we're at it these function calls could probably use some other cleanup. I think this non-idiomatic usage is partially caused by an overly complex flow of functions getting called.

Reply:
@Byron @spenserblack [1b682e9](https://github.com/o2sh/onefetch/pull/1335/commits/1b682e91aceb699e7e44fc739cc4e0fa8e486262) Did I do it right?
There seems to be a little more cloning involved.

Reply:
As a second follow up https://github.com/o2sh/onefetch/pull/1340